src/main/java/com/teragrep/jos_01/procfs/status/Cpuinfo.java

```diff
@@ -0,0 +1,381 @@
+                        processorFields.get("stepping"),
+                        processorFields.get("microcode"),
+                        processorFields.get("cpu MHz"),
+                        processorFields.get("cache size"),
+                        processorFields.get("physical id"),
+                        processorFields.get("siblings"),
+                        processorFields.get("core id"),
+                        processorFields.get("cpu cores"),
+                        processorFields.get("apicid"),
+                        processorFields.get("initial apicid"),
+                        processorFields.get("fpu"),
+                        processorFields.get("fpu_exception"),
+                        processorFields.get("cpuid level"),
+                        processorFields.get("wp"),
+                        processorFields.get("flags"),
+                        processorFields.get("vmx flags"),
+                        processorFields.get("bugs"),
+                        processorFields.get("bogomips"),
+                        processorFields.get("clflush size"),
+                        processorFields.get("cache_alignment"),
+                        processorFields.get("address sizes"),
+                        processorFields.get("power management"),
+                        processorFields
+                );
+                processors.add(processor);
+                processorFields = new LinkedHashMap<String, String>();
+                continue;
+            }
+            String[] keyValuePair = row.split(":");
+            String key = keyValuePair[0].trim();
+            String value;
+            if (keyValuePair.length == 1) {
+                value = "";
+            }
+            else {
+                value = keyValuePair[1].trim();
+            }
+            processorFields.put(key, value);
+        }
+    }
+
+    public ArrayList<Processor> processors() {
+        return processors;
+    }
+
+    public int cpuPhysicalCoreCount() {
+        Map<Integer, Integer> coreCounts = new HashMap<>();
+        for (Processor processor : processors) {
+            coreCounts.put(processor.physical_id, processor.cpu_cores);
+        }
+        int coreCount = 0;
+        for (Map.Entry<Integer, Integer> physicalCPU : coreCounts.entrySet()) {
+            coreCount = coreCount + physicalCPU.getValue();
+        }
+        return coreCount;
+    }
+
+    public int cpuCount() {
+        ArrayList<Integer> physicalIds = new ArrayList();
+        for (Processor processor : processors) {
+            physicalIds.add(processor.physical_id);
+        }
+        Map<Integer, Integer> distinct = new HashMap<>();
+        for (int physicalId : physicalIds) {
+            distinct.put(physicalId, distinct.getOrDefault(physicalId, 0) + 1);
+        }
+        return distinct.size();
+    }
+
+    public int cpuThreadCount() {
+        return processors.size();
+    }
+
+    public Map<String, String> statistics() {
```

Comment:
What happens if I run this function in a loop? Seems like the `statistics` will bloat until memory runs out?

Reply:
The size of statistics shouldn't increase with subsequent calls, since the keys of the Map don't change in \proc\cpuinfo. Subsequent calls would just replace the values for each pre-existing key.

Regardless, added a call to statistics.clear(), in case it's possible that the number of logical cpus changes in \proc\cpuinfo. This could lead to stale data being stuck in the Map.